client/app/src/screens/Results.js

```diff
@@ -1,12 +1,244 @@
 import React, { Component } from 'react';
-import { Text, View } from 'react-native';
+import { Platform, Text, TouchableOpacity, View } from 'react-native';
 import EStyleSheet from 'react-native-extended-stylesheet';
+import SideSwipe from 'react-native-sideswipe';
+import { AreaChart, BarChart, LineChart, StackedAreaChart } from 'react-native-svg-charts';
+import { curveNatural } from 'd3-shape';
+import { scaleTime } from 'd3-scale';
+import Constants from 'expo-constants';
+
+import { store } from '../state';
+import Layout from '../constants/Layout';
+import COLORS, { swatchColorInfo } from '../constants/Colors';
+import { CHOICE_COLOR, CHOICE_CHILLS } from '../constants/Choices';
+
+const COLOR_KEYS = Object.keys(swatchColorInfo);
+const COLOR_EMPTY = Object.fromEntries(COLOR_KEYS.map(color => [color, 0]));
+const CARD_MARGIN = 0.05 * Layout.window.width;
+const CARD_WIDTH = 0.9 * Layout.window.width;
+const CONTAINER_HEIGHT = Layout.window.height - Constants.statusBarHeight;
+const CARD_HEIGHT = 0.85 * CONTAINER_HEIGHT;
+const CHART_COLOR_AGGREGATE_HEIGHT = 0.75 * CARD_HEIGHT;
+const CHART_COLOR_USER_HEIGHT = 0.075 * CARD_HEIGHT;
+const CHART_CHILLS_HEIGHT = 0.8 * CARD_HEIGHT;
 
 class ResultsScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    const { songs } = props.navigation.state.params;
+    const { choices: userChoices } = store.getState();
+
+    this.songs = songs
+      .sort((a, b) => a.startTime.localeCompare(b.startTime))
+      .map(({ startTime, endTime, choices, choiceType, ...songInfo }) => {
+        const song = {
+          startTime,
+          endTime,
+          choices: choices.map(({ timestamp, ...choice }) => ({
+            timestamp: new Date(timestamp),
+            ...choice,
+          })),
+          choiceType,
+          chartProps: {
+            xAccessor: ({ item }) => item.timestamp,
+            xMax: new Date(endTime),
+            xMin: new Date(startTime),
+            xScale: scaleTime,
+            contentInset: {
+              top: Platform.select({
+                ios: 0,
+                android: 0.1 * CHART_COLOR_AGGREGATE_HEIGHT,
```

Comment:
One last ignorance question for this review: why the different handling here?

Reply:
> An object that specifies how much fake "margin" to use inside of the SVG canvas. This is particularly helpful on Android where overflow: "visible" isn't supported and might cause clipping.

https://github.com/JesperLekland/react-native-svg-charts/blob/dev/README.md#common-props